docs/provisioners/kustomize.md

```diff
@@ -0,0 +1,355 @@
+# kustomize Provisioner
+
+## Summary
+
+The `kustomize` provisioner is one of the [provisioners](https://github.com/operator-framework/rukpak/tree/main/internal/provisioner) of RukPak.
+It instantiates a given `kustomize+v0` bundle onto a cluster and then install the generated resources
+on the cluster.  The `kustomize+v0` bundle is a [Kustomization](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/).
+It does so by reconciling `Bundle` and `BundleDeployment` types that have
+the `spec.provisionerClassName` field set to `core-rukpak-io-kustomize`. This field must be set to the correct provisioner
+name in order for the `kustomize` provisioner to see and interact with the bundle.
+
+Supported source types for a helm bundle currently include the following:
+
+* A container image
+* A directory in a git repository
+* An [http](../sources/http.md) url
+* An [upload](../uploading-bundles.md) artifact
+
+Additional source types, such as a local volume are on the roadmap. These source types
+all present the same content, a directory containing a helm chart, in a different ways.
+
+### Install and apply a specific version of a `kustomize+v0` bundle
+
+> :warning: Anyone with the ability to create or update BundleDeployment objects can become a cluster admin. It's important
+> to limit access to this API via RBAC to only those that explicitly require access, as well as audit your bundles to
+> ensure the content being installed on-cluster is as-expected and secure.
+
+The `kustomize` provisioner can install and make available a specific `kustomize+v0` bundle in the cluster.
+
+Simply create a `BundleDeployment` resource that contains the desired specification of a Bundle resource.
+The `kustomize` provisioner will retrieve the referred kustomization files and instantiate the Bundle onto the cluster, and eventually install the generated resources
+ on the cluster.
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleDeployment
+metadata:
+  name: my-kustomize
+spec:
+  provisionerClassName: core-rukpak-io-kustomize
+  config:
+    path: dev
```

Comment:
It's entirely possible I glossed over the documentation section too quickly: can we explicitly call out the behavior of what happens when the spec.config.path has been populated in the BD vs. when it's been unspecified? Is the behavior here similar to the helm provisioner, where the referenced helm chart contents are the source of truth unless it's been overridden by the corresponding spec.config field that stores a values.yaml?

Reply:
Read through the changes further down. My understanding is that this is almost a required field to help expose which kustomize overlay should be used. Is that the right understanding? Does kustomize have the ability to identify a default overlay in the case that this field hasn't been configured?

Reply:
Your understanding is correct.  There is no default overlay so the `path` is required field.

Reply:
Maybe I'm missing some context, but the `kubectl kustomize` help output shows this as an example:
> \# Build the current working directory
> kubectl kustomize

Based on that, it seems like path should be optional, no?

Reply:
@joelanford you are right.  I tried it now.  It works like you described.

Reply:
> And if you combine that feature with something like bundle.spec.source.git.directory, then you could use a subdirectory of the git repo as the bundle root and then not specify a bd.config.path

I tried this but it didn't work because the git source just stores the files under the `bundle.spec.source.git.directory`. so  the `base` directory at the same level as the target directory doesn't exist or can not be referred.

Reply:
> so the base directory at the same level as the target directory doesn't exist or can not be referred.

I think I'm following this, and I'd say that we don't _need_ to land full support for all source types for the initial provisioner implementation as long as we're tracking the potential want/need to handle this more robustly in the future.